frontend/src/composables/useShootContext/index.js

```diff
@@ -904,12 +970,9 @@ export function createShootContextComposable (options = {}) {
             include: includeZones = [],
           } = {},
         } = provider
-        if (primary) {
-          dnsPrimaryProviderId.value = id
-        }
         let readonly = false
         if (!isNewCluster.value) {
-          const secret = findDnsProviderSecret(type, secretName)
+          const secret = findExtensionDnsProviderSecret(type, utils.getResourceRefName(initialManifest.value.spec?.resources, secretName))
```

Comment:
extract `utils.getResourceRefName(initialManifest.value.spec?.resources, secretName)` to variable